lib/src/kbasesearchengine/search/ObjectData.java

```diff
@@ -526,5 +548,16 @@ public Builder withKeyProperty(final String key, final String property) {
             keyProps.put(key, property);
             return this;
         }
+
+        /** Adds the highlight fields to the object.
+         * @param highlight the map of fields returned from elasticsearch.
+         * @return this builder.
+         */
+        public Builder withNullableHighlight(final Map<String, ArrayList> highlight) {
```

Comment:
Please follow the example of the other builders and don't add maps directly. Add key by key and enforce reasonable constraints on the key and value. In particular:
1) No nulls or empty strings
2) No unparameterized generics
3) Don't specify the List implementation

Reply:
so you mean separate it out into key and value? Can I directly add the ArrayList?

Reply:
I personally would probably do something like

```
public Builder withHighLight(final String field, final String highlight) {
    Utils.notNullOrEmpty(field, ...
    Utils.notNullOrEmpty(highlight, ...
    if (!higlights.containsKey(field)) {
        highlights.put(field, new HashSet<>());
    }
    highlights.get(field).add(highlight);
    return this;
}
```
I suppose you could do 
public Builder withHighLight(final String field, final List<String> highlights) {
    Utils.notNullOrEmpty(field, ...
    Utils.notNull(highlights, ...
    Utils.noNulls(highlights, ...
     this.highlights.put(field, highlights)
    return this;
 }

Reply:
Actually the latter implementation allows empty strings in the list